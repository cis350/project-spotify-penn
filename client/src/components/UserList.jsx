import React from 'react';
import {
  Space,
} from '@mantine/core';
import NewFriends from './NewFriendsTable';

function UserList() {
  return (
    <>
      <Space h="xl" />
      {/* <Title align="center"> Users </Title> */}
      <NewFriends />
      <Space h="40px" />
    </>
  );
}
export default UserList;
