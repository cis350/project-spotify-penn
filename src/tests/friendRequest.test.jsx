/* eslint-disable class-methods-use-this */
/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import UserList from '../components/UserList';

class ResizeObserver {
  observe() {}

  unobserve() {}

  disconnect() {}
}
window.ResizeObserver = ResizeObserver;

describe('Component rendering tests', () => {
  test('test that a user list is in the document', () => {
    // render the component
    const { getByRole } = render(<UserList />);
    // find the element by its role
    const newFriendList = getByRole('NewFriends');
    // assert that the element is in the document
    expect(newFriendList).toBeInTheDocument();
  });
  test('test Users header', () => {
    // render the component
    const { getByText } = render(<UserList />);
    // find the element by its role
    const titleText = getByText(/Users/);
    // assert that the element is in the document
    expect(titleText).toBeInTheDocument();
  });
});

test('snapshot test', () => {
  const component = renderer.create(<UserList />);
  const domTreeJSON = component.toJSON();
  expect(domTreeJSON).toMatchSnapshot();
});

test('click on a user profile -> lead to user profile page', async () => {
  render(<UserList />);
  const playlistElement = screen.queryByRole('Playlists');
  expect(playlistElement).toBeNull();

  await userEvent.click(screen.getByRole('NavLink'));
  expect(screen.getByRole('Playlists')).toBeInTheDocument();
});

test('Button change when friend is requested', async () => {
  render(<UserList />);
  const playlistElement = screen.queryByRole('Playlists');
  expect(playlistElement).toBeNull();

  await userEvent.click(screen.getByRole('NavLink'));
  expect(screen.getByRole('Playlists')).toBeInTheDocument();
  expect(screen.getByRole('Button')).toHaveTextContent('Add Friend');

  await userEvent.click(screen.getByRole('Button'));
  expect(screen.getByRole('Button')).toHaveTextContent('Requested');
});
