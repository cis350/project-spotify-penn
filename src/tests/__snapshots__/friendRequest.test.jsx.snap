// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Array [
  <div
    className="mantine-1cw9ra3"
  />,
  <h1
    className="mantine-Text-root mantine-Title-root mantine-44lm1w"
  >
     Users 
  </h1>,
  <div
    className="mantine-Center-root mantine-ojrz4j"
  >
    <div
      className="mantine-ScrollArea-root mantine-1deqy7c"
      dir="ltr"
      style={
        Object {
          "--radix-scroll-area-corner-height": "0px",
          "--radix-scroll-area-corner-width": "0px",
          "position": "relative",
        }
      }
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "[data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}",
          }
        }
      />
      <div
        className="mantine-b6zkvl mantine-ScrollArea-viewport"
        data-radix-scroll-area-viewport=""
        style={
          Object {
            "overflowX": "scroll",
            "overflowY": "scroll",
          }
        }
      >
        <div
          style={
            Object {
              "display": "table",
              "minWidth": "100%",
            }
          }
        >
          <table
            className="mantine-Table-root mantine-z8tlxi"
          >
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="mantine-13xtcol"
  />,
]
`;
